src/hera/shared/serialization.py

```diff
@@ -31,14 +26,10 @@ def default(self, o: Any):
         # Note that these are slightly different outputs b/w v1 and v2
         # v1 will give the actual python object whereas v2 will serialize it into
         # a json compatible format.
-        if _PYDANTIC_VERSION == 1:
-            if isinstance(o, BaseModel):
-                return o.dict(by_alias=True)
-        else:
-            if isinstance(o, BaseModel):
-                return o.model_dump(by_alias=True, mode="json")  # type: ignore
-            if isinstance(o, V1BaseModel):
-                return o.dict(by_alias=True)
+        if isinstance(o, V1BaseModel):
+            return o.dict(by_alias=True)
+        if isinstance(o, BaseModel):
+            return o.model_dump(by_alias=True, mode="json")  # type: ignore
```

Comment:
Not sure if it's to be trusted in this case, but I tried removing the `ignore` and `make lint` still passed
```suggestion
            return o.model_dump(by_alias=True, mode="json")
```